src/aggregate.d

```diff
@@ -270,13 +276,15 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             return 0;
         }
 
-        fields.setDim(0);
-
         for (size_t i = 0; i < members.dim; i++)
         {
             auto s = (*members)[i];
             if (s.apply(&func, cast(void*)this))
+            {
+                if (sizeok != SIZEOKnone)
+                    return true;
```

Comment:
Typical LDC not upstreaming fixes. :-)

@rainers - it would be a different ICE, but looks from the same symptom.